app/views/OfferedCourses/Courses.js

```diff
@@ -14,22 +14,22 @@ class Courses extends Component {
     }
     async componentDidMount() {
         const courses = await ajax.fetchInitialCourses();
-        this.setState(prevState => {
-            return { courses };
-        });
-        console.log(courses);
+        this.setState({ courses });
+        console.log('COURSES' + courses);
     }
 
 
     render() {
 
-        //console.log(this.state.courses);
-        //console.log(Object.keys(this.state.courses).length);
+        console.log('STATE COURSES ');
+        console.log(this.state.courses.data);
+        console.log('COURSES LENGTH' + Object.keys(this.state.courses).length);
+        console.log('COURSES LENGTH' + Object.keys(this.state.courses).length);
 
         return (
             <View style={styles.container}>
-                {this.state.courses.length > 0 ? (
-                    <CoursesList courses={this.state.courses}/>
+                {Object.keys(this.state.courses).length > 0  ? (
+                    <CoursesList courses={this.state.courses.data}/>
```

Comment:
Why is this component needed? Couldn't we implement the functionality in this component in courseList.js? Or maybe rename this file to "fetchCourses.js" or something like that.

Reply:
This component is made for fetching courses from api and connected to the ajax.js. The choice for separating it was my personal and I prefer to make components small and reusable